locksmith/src/operations/subscriptionOperations.ts

```diff
@@ -0,0 +1,150 @@
+import networks from '@unlock-protocol/networks'
+import {
+  SubgraphService,
+  Web3Service,
+  getAllowance,
+  getErc20BalanceForAddress,
+  getErc20Decimals,
+  getErc20TokenSymbol,
+} from '@unlock-protocol/unlock-js'
+import { ethers } from 'ethers'
+import { KeySubscription } from '../models'
+import { Op } from 'sequelize'
+
+import dayjs from 'dayjs'
+import relative from 'dayjs/plugin/relativeTime'
+import duration from 'dayjs/plugin/duration'
+
+dayjs.extend(relative)
+dayjs.extend(duration)
+
+interface Amount {
+  amount: string
+  decimals: number
+  symbol: string
+}
+
+export interface Subscription {
+  next: number | null
+  balance: Amount
+  price: Amount
+  possibleRenewals: string
+  approvedRenewals: string
+  type: 'Crypto' | 'Stripe'
+}
+
+interface GetSubscriptionsProps {
+  tokenId: string
+  lockAddress: string
+  owner: string
+  network: number
+}
+
+export const getSubscriptions = async ({
+  tokenId,
+  lockAddress,
+  owner,
+  network,
+}: GetSubscriptionsProps): Promise<Subscription[]> => {
+  const subgraphService = new SubgraphService(networks)
+  const key = await subgraphService.key(
```

Comment:
I don't think we should use the subgraph here. It is too britle IMO.
Let's retrieve the fields we need from it? `keyPrice` and `tokenAddress` and `version` ?

Reply:
for now use of subgraph because we don't have the we to get specific params from `getLock`